From afaa8b0cef7da4354b7fab012e3533c7d39f08b5 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@skynet.be>
Date: Fri, 31 Mar 2017 15:18:17 +0200
Subject: [PATCH 1/1] Use a stateless configuration

Update of previous patch from Michael Meinertzhagen <mmeinert@protonmail.com>
---
 src/Makefile.am | 2 +-
 src/rcfile.c    | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index bada082..5e163fb 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\" \
+AM_CPPFLAGS = -DDATADIR=\"$(datadir)\" -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\" \
 	-I$(top_builddir)/lib -I$(top_srcdir)/lib
 
 CLEANFILES = revision.h
diff --git a/src/rcfile.c b/src/rcfile.c
index e751d86..e108751 100644
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -1265,7 +1265,11 @@ void parse_one_nanorc(void)
 /* First read the system-wide rcfile, then the user's rcfile. */
 void do_rcfiles(void)
 {
-    nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    if (access(SYSCONFDIR "/nanorc", F_OK) == 0) {
+		nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    } else {
+		nanorc = mallocstrcpy(nanorc, DATADIR "/defaults/nano/nanorc");
+    }
 
     /* Process the system-wide nanorc. */
     parse_one_nanorc();
-- 
2.12.2

